### ISSUE TITLE
Flutter test doesn't report the failing tests in the summary

### ISSUE THREAD

**Issue by gaaclarke** [flutter/flutter#102387]
Often times it's difficult to know which tests have failed when `flutter test` runs.  For example: https://cirrus-ci.com/task/6163921269686272?logs=main#L3250

The report at the end just says:
```
07:49 +6413 ~14 -5: Some tests failed.
```

Then you have to scroll up looking for the failure which isn't always easy to find and if you search for "fail" in the text many tests print out that text while not failing.

The report at the end should list the tests that failed.

**Comment by gaaclarke** [comment]
In the case of cirrus every once in a while I forget that the output is abbreviated.  It isn't so hard to find the failing error if you have all the output.

**Comment by christopherfujino** [comment]
@gaaclarke I agree!

**Comment by xiprox** [comment]
Having this issue in a different scenario. Recently upgraded to Flutter 3.0 and getting warnings (63 lines) from some dependencies for each test run. By the time my tests finish executing, my terminal can no longer display the failed tests. Need to then re-run them in VS Code.

**Comment by dmrickey** [comment]
I agree it's currently not ideal to have to scroll through hundreds of tests to see which ones failed. We use `-r expanded` right now because we need the individual failures. But it would be super nice if there was a separate option that collapsed the successes but kept the failures individualized.

**Comment by GP4cK** [comment]
I worked around it by doing `flutter -r json` and then parse the output with [jq](https://jqlang.github.io/jq/) to get all the file names with at least one failing test.

It was my first time using jq so maybe there's a better way to do it but this is how it looks:

```bash
flutter test -r json "${files[@]}" | jq -s -r '
[
  .[]
  | select(.result != null and ([.result] | inside(["error", "failure"])))
  | .testID
] as $testIDs
|
[
  .[]
  | select(.test.id != null and ([.test.id] | inside($testIDs)))
  | .test.suiteID
] as $suiteIDs
|
  .[]
  | select(.suite.id != null and ([.suite.id] | inside($suiteIDs)))
  | .suite.path
  | match("test/.*").string
'
```
